quickwit-serve/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2021 Quickwit Inc.
+ *
+ * Quickwit is offered under the AGPL v3.0 and as commercial software.
+ * For commercial licensing, contact us at hello@quickwit.io.
+ *
+ * AGPL:
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+mod args;
+mod grpc_adapter;
+
+use std::sync::Arc;
+
+pub use args::ServeArgs;
+use quickwit_metastore::MetastoreUriResolver;
+use quickwit_search::SearchServiceImpl;
+use quickwit_storage::StorageUriResolver;
+use tracing::debug;
+mod error;
+mod rest;
+
+pub use crate::error::ApiError;
+
+pub async fn serve_cli(args: ServeArgs) -> anyhow::Result<()> {
```

Comment:
Maybe we can refactor by following what we have been doing. create `serve_cli` in the `quickwit-cli` crate to allow it to display user messages like.:
```bash 
Server running on http://localhost:8000  
You can search with `curl http://localhost:8000/api/v1/search?query=ada`
```
Also the server start even with a wrong metastore-uri.
we could handle this in smaller tasks though.